cape-char.el

```diff
@@ -28,41 +28,52 @@
 
 (autoload 'thing-at-point-looking-at "thingatpt")
 
+(defmacro cape-char--ensure-str (char-or-str)
+  "IF-expression to convert CHAR-OR-STR to string only if it's a char"
+  `(if (characterp ,char-or-str)
+      (char-to-string ,char-or-str) ,char-or-str))
+
+(defmacro cape-char--ensure-char (char-or-str)
+  "IF-expression to convert CHAR-OR-STR to char only if it's a string"
+  `(if (stringp ,char-or-str)
+      (string-to-char ,char-or-str) ,char-or-str))
+
 ;; Declare as pure function which is evaluated at compile time. We don't use a
 ;; macro for this computation since packages like `helpful' will
 ;; `macroexpand-all' the expensive `cape-char--define' macro calls.
 (eval-when-compile
```

Comment:
I hope we can remove the `eval-when-compile` here and just move the defun to the toplevel as a normal function. Also the (incorrect) purity declaration and the comment can go away.

Reply:
[1496fa1](https://github.com/minad/cape/pull/91/commits/1496fa1ad46c99e7dfa2ed2182309d53a801e263)